### PR TITLE
[FLINK-27111][table][docs] Update docs regarding TableEnvironment configuration

### PR DESCRIPTION
**Pull request by matriv**
## What is the purpose of the change

Following the work on [FLINK-16835] update the docs accordingly to
mention the use of `EnvironmentSettings` and that `TableEnvironement`
related configuration can also be set in `flink-conf.yaml`.

## Brief change log

  - Update docs for `TableEnvironment` configuration
  - Update examples for instantiating `TableEnvironment` using `EnvironmentSettings`.

## Verifying this change

This change is only docs related without the need of any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented? **update docs**

### CODE REVIEW

File: docs/content/docs/dev/table/config.md [hunk 1]
```diff
@@ -96,6 +124,11 @@ Flink SQL> SET 'table.exec.mini-batch.size' = '5000';
 {{< /tab >}}
 {{< /tabs >}}
 
+**Note:** All of the following configuration options can also be set globally in
```
Comment (slinkydeveloper) [inline docs/content/docs/dev/table/config.md:127]: Just a typography comment here: Use the hints here: https://github.com/apache/flink/tree/master/docs#info-hints

Comment (matriv) [reply docs/content/docs/dev/table/config.md:127]: Thank you!

**Comment by matriv** [comment]
@flinkbot run azure

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Pull request merged**